### PR TITLE
Feat41 & Feat57 - YouTube video / styling changes / component re-work(v2)

### PR DESCRIPTION
**Pull request by JimmyLee87**

### CODE REVIEW

File: client/actions/index.js [hunk 1]
```diff
@@ -28,10 +31,23 @@ export function fetchCandidateIndustryContributors(crpid){
   }
 }
 
+// fetch twitter data based on twitter candidate nickname
+export function fetchTwitter(cid){
+  const route = 'twitter?candtwitternickname=';
+  const url = `${URL}${route}${cid}`
+  const request = axios.get(url);
+  request.then((data) => {   
+  })
```
Comment (ryannewton) [inline client/actions/index.js]: Do we still need this request.then do?

File: client/components/detailed_profile.js [hunk 2]
```diff
@@ -26,72 +40,82 @@ class DetailedProfile extends Component {
...
+        break;
+      case 'I':
+        logo = ind;
+        break;
+      case 'L':
+        logo = lib;
+        break;
+      case 'G':
+        logo = gree
+        break;n;
```
Comment (ryannewton) [inline client/components/detailed_profile.js]: left 'n;' on this line

File: client/components/detailed_profile.js [hunk 3]
```diff
@@ -26,72 +40,82 @@ class DetailedProfile extends Component {
...
+      case 'N':
+        logo = none
+        break;;
```
Comment (ryannewton) [inline client/components/detailed_profile.js]: have an extra ;

File: client/containers/candidate_finance.js [hunk 4]
```diff
@@ -41,12 +40,16 @@ class CandidateFinance extends Component {
...
+        <div className="candidate-finance-tiles">{this.renderCandidateFinance()}</div>
       </div>
     );
   }
 }
 
+function mapDispatchToProps(dispatch){
+  return bindActionCreators({ fetchCandidateIndustryContributors }, dispatch);
+}
+
```
Comment (ryannewton) [inline client/containers/candidate_finance.js]: It doesn't look like this function is being used.

File: client/containers/profiles_list.js [hunk 5]
```diff
@@ -2,8 +2,22 @@ import _ from 'lodash';
...
+class ProfilesList extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.backButtonClick = this.backButtonClick.bind(this);
+  }
+
+  backButtonClick() {
+    console.log('backbuttonfiring in profiles list');
```
Comment (ryannewton) [inline client/containers/profiles_list.js]: Do we still need this console.log?

File: client/containers/search_bar.js [hunk 6]
```diff
@@ -24,11 +24,15 @@ class SearchBar extends Component {
...
-
 		this.props.fetchByZip( this.state.searchTerm );
+		this.props.showList();
 		this.setState( { searchTerm: '' } );
 	}
 
+	onButtonClick(event) {
+		// this.props.showList();
+	}
+
```
Comment (ryannewton) [inline client/containers/search_bar.js:37]: Is this a placeholder for future functionality?
If so, let's describe the planned function in the comments

File: client/containers/twitter.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+import React, { Component } from 'react';
+import { fetchTwitter, fetchCandidate } from '../actions/index';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+class Twitter extends Component { 
+
+  componentWillMount() {
+    this.props.fetchCandidate(this.props.candId)  //undefined
+    .then ((data) => {
```
Comment (ryannewton) [inline client/containers/twitter.js:11]: What does the undefined comment mean?

File: client/containers/twitter.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      this.props.fetchTwitter(data.payload.data['0'].twitter_id)
+      .then ((data) => {
+      })
```
Comment (ryannewton) [inline client/containers/twitter.js]: Do we need this last .then statement?

File: client/routes.js [hunk 9]
```diff
@@ -3,15 +3,14 @@ import React from 'react';
 import { Route, IndexRoute } from 'react-router';
 
 import App from './components/app';
-import ProfilesList from './containers/profiles_list';
-import SearchBar from './containers/search_bar';
+import LandingPage from './components/landing_page';
+// import SearchBar from './containers/search_bar';
```
Comment (ryannewton) [inline client/routes.js]: Do we need this comment still?

File: server/main.js [hunk 10]
```diff
@@ -32,6 +34,29 @@ app.route('/candname')
...
+                    obj.created_at = '';
+                    obj.text = '';
+                    obj.user = '';
+                    obj.location = '';
+                    obj.followers = '';
+                    obj.url = '';
+
+                    var array=[];
+                    array.push(obj);
+                  res.status(200).send(array);
```
Comment (ryannewton) [inline server/main.js:54]: Perhaps use us a 400 status code ("Bad request")

File: server/main.js [hunk 11]
```diff
@@ -69,6 +94,7 @@ app.route('/candCampAddress').get(function (req, res) {
 app.route('/candbio').get(function (req, res) {
   var candbio = req.query.candId;
   vs.collectCandidateDetails(candbio).then(function (results) {
+    console.log(results);
```
Comment (ryannewton) [inline server/main.js]: Do we still need this console.log?
If so, let's add some context before the variable.

File: client/actions/index.js (hunk 1 above)
Comment (JimmyLee87) [inline client/actions/index.js]: cleared

File: client/components/detailed_profile.js (hunk 2 above)
Comment (JimmyLee87) [inline client/components/detailed_profile.js]: cleared

File: client/components/detailed_profile.js (hunk 3 above)
Comment (JimmyLee87) [inline client/components/detailed_profile.js]: cleared

File: client/containers/candidate_finance.js (hunk 4 above)
Comment (JimmyLee87) [inline client/containers/candidate_finance.js]: cleared

File: client/containers/profiles_list.js (hunk 5 above)
Comment (JimmyLee87) [inline client/containers/profiles_list.js]: cleared

File: client/containers/search_bar.js (hunk 6 above)
Comment (JimmyLee87) [inline client/containers/search_bar.js:37]: added notes - for the eventual geo-location option (not functioning at the moment...)

File: client/containers/twitter.js (hunk 7 above)
Comment (JimmyLee87) [inline client/containers/twitter.js:11]: cleared

File: client/containers/twitter.js (hunk 8 above)
Comment (JimmyLee87) [inline client/containers/twitter.js]: nope - cleared

File: client/routes.js (hunk 9 above)
Comment (JimmyLee87) [inline client/routes.js]: cleared

File: server/main.js (hunk 10 above)
Comment (JimmyLee87) [inline server/main.js:54]: allows us to populate the field even with no response (from the server side). 

We can discuss a refactor during code cleanup week.

File: server/main.js (hunk 11 above)
Comment (JimmyLee87) [inline server/main.js]: cleared

**Pull request merged**